### PR TITLE
RUST-960 Remove the `decimal128` feature flag

### PR DESCRIPTION
**Pull request by patrickfreed**
RUST-960

This PR removes the `decimal128` feature flag and associated functionality. See [this comment](https://github.com/mongodb/bson-rust/issues/282#issuecomment-889958970) for more context on the reasoning behind this.

As part of this, `Decimal128::from_bytes` and `Decimal128::bytes` were introduced, as well as `Document::get_decimal128` and `Document::get_decimal128_mut`. The derived `PartialOrd` implementation was also removed, as it was not correct without the feature flag.

### CODE REVIEW

**Review by abr-egn — APPROVED** [review]

File: .evergreen/run-tests-serde.sh [hunk 1]
```diff
@@ -1,7 +0,0 @@
-#!/bin/sh
```
Comment (isabelatkinson) [inline .evergreen/run-tests-serde.sh:1]: Why was this file removed?

Comment (patrickfreed) [reply .evergreen/run-tests-serde.sh:1]: In #279, I updated `run-tests.sh`, `run-tests-u2i.sh`, and `run-tests-decimal128.sh` to each run the serde tests to cover our feature flags, so this script wasn't actually being used anymore.

**Review by isabelatkinson — APPROVED** [review]

**Review by NBSquare — APPROVED** [review]
